### PR TITLE
[lstm] make lstm_core as class

### LINKED ISSUE #1997: make lstm_core function as abstract class
For now lstm_core is consist of functions.
Make this as abstract class and lstm/lstm_cell/zoneout_lstmcell should inherit this class.

### PR DESCRIPTION
**Pull request by lhs8928**
- Make lstm_core as class so let lstm/lstm_cell/zoneout_lstmcell inherit this.

close #1997

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

**Review by DonghakPark — APPROVED** [review]
LGTM!

**Review by songgot — APPROVED** [review]
LGTM

File: nntrainer/layers/lstmcell_core.h [hunk 1]
```diff
@@ -16,91 +16,152 @@
...
-void lstmcell_forwarding(const unsigned int batch_size, const unsigned int unit,
-                         const bool disable_bias, const bool integrate_bias,
-                         ActiFunc &acti_func, ActiFunc &recurrent_acti_func,
-                         const Tensor &input, const Tensor &prev_hidden_state,
-                         const Tensor &prev_cell_state, Tensor &hidden_state,
-                         Tensor &cell_state, const Tensor &weight_ih,
-                         const Tensor &weight_hh, const Tensor &bias_h,
-                         const Tensor &bias_ih, const Tensor &bias_hh,
-                         Tensor &ifgo);
+class LSTMCore {
```
Comment (jijoongmoon) [inline nntrainer/layers/lstmcell_core.h:28]: how about LSTMCore :: public LayerImpl()??

File: nntrainer/layers/lstmcell.cpp [hunk 2]
```diff
@@ -32,13 +31,7 @@ enum LSTMCellParams {
...
 LSTMCellLayer::LSTMCellLayer() :
   LayerImpl(),
-  lstmcell_props(props::Unit(), props::IntegrateBias(),
-                 props::HiddenStateActivation() = ActivationType::ACT_TANH,
-                 props::RecurrentActivation() = ActivationType::ACT_SIGMOID,
-                 props::DropOutRate()),
-  acti_func(ActivationType::ACT_NONE, true),
-  recurrent_acti_func(ActivationType::ACT_NONE, true),
-  epsilon(1e-3) {
+  lstmcell_props(props::DropOutRate()) {
```
Comment (jijoongmoon) [inline nntrainer/layers/lstmcell.cpp:34]: so, only dropout rate is the property for LSTMCell? and others ( Activation for Hidden and Recurrent & epsilon ) are for the LSTMCore? Could you add some comment about it?

File: nntrainer/layers/lstmcell_core.h (hunk 1 above)
Comment (lhs8928) [reply nntrainer/layers/lstmcell_core.h]: Done

File: nntrainer/layers/lstmcell.cpp (hunk 2 above)
Comment (lhs8928) [reply nntrainer/layers/lstmcell.cpp]: Added

**Pull request merged**